app/main.go

```diff
@@ -123,6 +129,11 @@ func main() {
 		log.Fatalf("Invalid flags: %v", err)
 	}
 
+	if *enableLocalCPUProfiler {
+		log.Debugf("CPU Profiler enabled.")
+		defer profile.Start(profile.CPUProfile, profile.ProfilePath(".")).Stop()
```

Comment:
My understanding of defer, is that the function call isn't actually executed till main() exits.

I might be wrong here - but would this mean the profiler will only start once main exits, and then stop right after?

Reply:
The documentation for this package states that this is the correct usage: https://github.com/pkg/profile

I believe that the profile.Start() is called immediately but .Stop() is not executed til main() exits. The profile produced with this code records the correct runtime duration so it seems to work as expected.